docs/source/release.rst

```diff
@@ -2,152 +2,192 @@
 
 .. _release:
 
-
-================================================
-Core Developers' Reference: Making a New Release
-================================================
+================================
+Releasing a new version of Mitiq
+================================
 
 .. note::
-	These instructions are aimed at the mantainers of the ``mitiq`` library.
-
-When the time is ready for a new release, follow the checklist and instructions of this document to go through all the steps below:
+    These instructions are aimed at the mantainers of the ``mitiq`` library.
 
-.. contents:: :local:
-    :depth: 3
+When the time is ready for a new release, follow the checklist and
+instructions of this document to go through all the steps below:
 
-
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Make sure that the changelog is updated
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-Make sure that the changelog is updated at ``mitiq/CHANGELOG.md`` and if not, add the latest merged pull requests (PRs), including author and PR number (@username, gh-xxx).
+.. contents::
+   :local:
+   :depth: 3
 
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
 Work in a siloed environment
 ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
-- Create a conda environment
+
+It is recommended that the release is performed in a new, clean virtual
```

Comment:
Should we make a docker container with this env setup to isolate things even more (if we don't have one already)?

Reply:
Same here -- I don't think this is a blocker for this PR, but we might consider creating a Docker image with the standard mitiq setup (and maybe jupyter) so that people can play around with the tool without installing. I don't think it gives us much lift for CI/CD though, as setting up a python env on GH actions is super easy.